File: my_cpp_pkg/src/reset_counter.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/set_bool.hpp"

using namespace std::chrono_literals;

 
int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<rclcpp::Node>("reset_counter");

    auto client = node->create_client<example_interfaces::srv::SetBool>("reset_counter");
    while (!client->wait_for_service(1s))
    {
        RCLCPP_WARN(node->get_logger(), "Waiting for the server ...");
    }

    auto request = std::make_shared<example_interfaces::srv::SetBool::Request>();
    request->data = true;

    auto future = client->async_send_request(request);
    rclcpp::spin_until_future_complete(node, future);
    auto response = future.get();
    RCLCPP_INFO_STREAM(node->get_logger(), "success: " 
        << std::boolalpha << response->success << ", message: " << response->message);  
    
    rclcpp::shutdown();
    return 0;
}

File: my_py_pkg/my_py_pkg/reset_counter.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.srv import SetBool


def main(args=None):
    rclpy.init(args=args)
    node = Node("reset_counter")

    client = node.create_client(SetBool, "reset_counter")
    while not client.wait_for_service(timeout_sec=1.0):
        node.get_logger().warn("Waiting for number_counter server ...")

    request = SetBool.Request()
    request.data = True

    future = client.call_async(request)
    rclpy.spin_until_future_complete(node, future)

    response = future.result()
    node.get_logger().info("\nCounter reset: %s\nmessage: %s" % (response.success, response.message))
    
    rclpy.shutdown()
 
 
if __name__ == "__main__":
    main()